src/wallet/wallet.cpp

```diff
@@ -2554,26 +2540,35 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             return false; // TODO: Allow non-wallet inputs
     }
 
+    // Return early if no other inputs can be selected or we selected enough
+    if ((coin_control.HasSelected() && !coin_control.fAllowOtherInputs)
+            || (nValueFromPresetInputs >= nTargetValue)) {
+        nValueRet = nValueFromPresetInputs;
+        setCoinsRet = setPresetCoins;
+        return (nValueRet >= nTargetValue);
+    }
+
     // remove preset inputs from vCoins
-    for (std::vector<COutput>::iterator it = vCoins.begin(); it != vCoins.end() && coin_control.HasSelected();)
-    {
-        if (setPresetCoins.count(CInputCoin(it->tx->tx, it->i)))
+    for (std::vector<COutput>::iterator it = vCoins.begin(); it != vCoins.end() && coin_control.HasSelected();) {
```

Comment:
nit: this could be a const_iterator

Reply:
no, it's deleting entries?

Reply:
The (non-const) vector is deleting entries, the COutputs themselves are unmodified.

Reply:
ah, I assume const_iterator was const for everything, including the underlying vector. TIL